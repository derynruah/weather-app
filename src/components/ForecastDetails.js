import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, humidity, wind, temperature } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__humidity">{humidity}%</div>
      <div className="forecast-details__wind-speed">{wind.speed}</div>
      <div className="forecast-details__wind-direction">{wind.direction}</div>
      <div className="forecast-details__max-temp">{temperature.max}&deg;C</div>
      <div className="forecast-details__min-temp">{temperature.min}&deg;C</div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
