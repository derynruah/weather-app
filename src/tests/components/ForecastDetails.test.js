import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    forecast: {
      date: 1111111,
      humidity: 10,
      temperature: {
        min: 12,
        max: 18,
      },
      wind: {
        speed: 24,
        direction: "n",
      },
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails forecast={validProps.forecast} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(
      <ForecastDetails forecast={validProps.forecast} />
    );

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("10%")).toHaveClass("forecast-details__humidity");
    expect(getByText("12°C")).toHaveClass("forecast-details__min-temp");
    expect(getByText("18°C")).toHaveClass("forecast-details__max-temp");
    expect(getByText("24")).toHaveClass("forecast-details__wind-speed");
    expect(getByText("n")).toHaveClass("forecast-details__wind-direction");
  });
});
