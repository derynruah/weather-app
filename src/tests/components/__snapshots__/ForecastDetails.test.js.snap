// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__humidity"
    >
      10%
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      24
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      n
    </div>
    <div
      class="forecast-details__max-temp"
    >
      18°C
    </div>
    <div
      class="forecast-details__min-temp"
    >
      12°C
    </div>
  </div>
</DocumentFragment>
`;
